answerking_app/services/order_service.py

```diff
@@ -37,26 +28,26 @@ def get_by_id(order_id: str) -> Order | None:
 
 
 def create(body: OrderCreateDict) -> Order | None:
-    try:
-        address: str = validate_address_string(body["address"])
-    except (KeyError, ValidationError):
+    serialized_order = ClientOrderSerializer(data=body)
+    if not serialized_order.is_valid():
         return None
-
     status, _ = Status.objects.get_or_create(status="Pending")
 
-    created_order: Order = Order(address=address, status=status, total=0.00)
+    created_order: Order = Order(
+        address=serialized_order.data["address"], status=status, total=0.00
+    )
     created_order.save()
 
     try:
-        order_items: list[OrderItems] = body["order_items"]
-        if not order_items:
+        if not serialized_order.data["order_items"]:
             return created_order
     except KeyError:
         return created_order
 
-    for oi in order_items:
+    for oi in serialized_order.data["order_items"]:
         try:
             try:
+                # note for tommorrow, io doesn not exist in the oi object
```

Comment:
I know this is not your code Pietro, but shouldn't we keep try except blocks short rather than put everything inside. So here `item: Item = Item.objects.get(pk=oi["id"])` should be the only thing in the block for the exception `Item.DoesNotExist`

Reply:
yes, I agree, it should usually be a single expression per block